config/pipeline.yaml

```
@@ -342,9 +369,86 @@ platforms:
 
   shell:
 
+  kontron-kswitch-d10-mmt-6g-2gs:
+    <<: *armel-device
+    mach: microchip
+    dtb: dtbs/lan966x-kontron-kswitch-d10-mmt-6g-2gs.dtb
+
+  kontron-kswitch-d10-mmt-8g:
+    <<: *armel-device
+    mach: microchip
+    dtb: dtbs/lan966x-kontron-kswitch-d10-mmt-8g.dtb
+
+  imx6dl-riotboard:
+    mach: imx
+    class: arm-dtb
+    boot_method: barebox
+    rules:
+      defconfig:
+        - '!imx_v4_v5_defconfig'
+        - '!multi_v5_defconfig'
+        - '!allmodconfig'
+
+  imx6qp-wandboard-revd1:
+    mach: imx
+    class: arm-dtb
+    boot_method: barebox
+    rules:
+      defconfig:
+        - '!imx_v4_v5_defconfig'
+        - '!multi_v5_defconfig'
+        - '!allmodconfig'
+
+  dove-cubox:
+    mach: mvebu
+    class: arm-dtb
```

```suggestion
    <<: *armel-device
    mach: marvell
    dtb: dtbs/filename.dtb
```

`class` isn't handled in the new system, and `mach` should match the vendor's DTS folder name.

Maybe creating an `armel-barebox-device` anchor would make sense here.